atc/db/build.go

```diff
@@ -1216,6 +1216,11 @@ func (b *build) SaveOutput(
 	versionJSON := string(versionBytes)
 
 	if newVersion {
+		err = theResource.(*resource).setResourceConfigScopeInTransaction(tx, resourceConfigScope)
```

Comment:
I think in practice what you have here will work, but I think it'd be a slight improvement to make this unconditional, and to run it after `incrementCheckOrder`.

Technically the order of operations shouldn't make a difference, since it all runs in a transaction, but in `CheckStep` we only point the resource to the scope after the versions are saved in the proper order, and it matters in that scenario, so it would be nice for this to be consistent just so it doesn't raise questions for someone reading the code.

As for making it conditional, I just don't think it needs to be, and it'd be simpler to think of `put` and `check` the same way (determine config, save versions if necessary, point resource to config).

Reply:
Sure, fixed.